### PR TITLE
fix: improve sd-textarea a11y [skip chromatic]

### LINKED ISSUE #1492: feat[dev]: ✨ implement A11y improvements to sd-textarea
## Description
This task describes the Storybook implementation for the sd-textarea component as part of the A11y improvements outlined in Epic https://github.com/solid-design-system/solid/issues/1465.

A11y improvements for sd-textarea can be found here: [sd-textarea Notes](https://www.notion.so/matuzo/sd-textarea-10cf3f6e1bc08074b03ddb94da4f439b).

----------------------

## Comment
**Problem**: Invalid fields don't semantically indicate that they're invalid.
**Solution:** Put `aria-invalid="true"` on invalid fields.

**Problem:** The error message in invalid fields is not connected to the respective field.
**Solution:** Use `aria-describedby`, just like you're doing it with the help text.
> Fix this in every input component. 

```
<textarea aria-describedby="help-text invalid-message"></textarea>
<div id="help-text">Help text</div>
<div id="invalid-message">Error message</div>
```

<details>
<summary>
For Design
</summary>
## Notes

- Please note that `maxlength` doesn't indicate the maximum number of characters to screen readers. If there's no help text, it can be confusing.

- Consider noting in the docs that the placeholder is not a replacement for the label but only an addition.

- Disabled textareas: See [@sd-button](https://www.notion.so/sd-button-104f3f6e1bc0800b9f72d2a046316ec3?pvs=24). Everything I wrote there about disabling form elements applies here as well.

</details>

## Related WCAG criteria
- https://www.w3.org/WAI/WCAG22/quickref/#labels-or-instructions

### Open Questions
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by paulovareiro29**
## Description:

Closes #1492 

Goals:
- [x] Implement `aria-invalid` attribute on textarea when the field is invalid.
- [x] Implement `invalid-message` on `aria-describedby` 

## Definition of Reviewable:

- [x] E2E tests (features, a11y, bug fixes) are created/updated

- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/textarea/textarea.ts [hunk 1]
```diff
@@ -404,7 +405,8 @@ export default class SdTextarea extends SolidElement implements SolidFormControl
               spellcheck=${ifDefined(this.spellcheck)}
               enterkeyhint=${ifDefined(this.enterkeyhint)}
               inputmode=${ifDefined(this.inputmode)}
-              aria-describedby="help-text"
+              aria-describedby="help-text invalid-message"
+              ?aria-invalid=${isInvalid}
```
Comment (smfonseca) [inline packages/components/src/components/textarea/textarea.ts:409]: I think we should set the value to be in conformity with the [ARIA specification](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-invalid)

**Comment by paulovareiro29** [comment]
Some fields are invalid even when not touched, e.g. required or min-length fields are invalid when not touched, since it is empty, but in this cases the aria-invalid should not show, since the user might still be working to fill in the field.

Now `aria-invalid="true"` only shows when the field in is also a visual invalid state. This is controlled by `showInvalidStyle` attribute.

**Review by Vahid1919 — DISMISSED** [review]

**Review by smfonseca — APPROVED** [review]

**Review by karlbaumhauer — DISMISSED** [review]

**Comment by paulovareiro29** [comment]
The previous change broke the tests I had done, just pushed a change to fix the tests...
It is rather annoying that the tests quality gate is always falling.

**Review by smfonseca — APPROVED** [review]

**Comment by MartaPintoTeixeira** [comment]
the margin between the help-text and the text area and the margin between the error message and the text area is not consistent. will create a ticket to solve this issue in both figma and dev

**Comment by MartaPintoTeixeira** [comment]
Where it says: 
Related templates:
Tooltip

change it to: 
Related templates:
Textarea with Tooltip

**Comment by MartaPintoTeixeira** [comment]
for the invalid sample instead of "Error message" lets have "Error text" as in figma

**Comment by MartaPintoTeixeira** [comment]
for the disabled section the content inside the textarea should be neutral-500 and is neutral-700.
let me know if this is an issue in the component. if so please open a ticket to fix it and align with figma.

**Comment by paulovareiro29** [comment]
@MartaPintoTeixeira Issue has been created to address the disabled text color #1661

**Review by MartaPintoTeixeira — APPROVED** [review]

**Pull request merged**